libtextsecure/outgoing_message.js

```diff
@@ -115,22 +194,24 @@ OutgoingMessage.prototype = {
     this.errors[this.errors.length] = error;
     this.numberCompleted();
   },
-  reloadDevicesAndSend(number, recurse) {
+  reloadDevicesAndSend(primaryPubKey) {
     const ourNumber = textsecure.storage.user.getNumber();
-    return () =>
+
+    return (
       libloki.storage
-        .getAllDevicePubKeysForPrimaryPubKey(number)
+        .getAllDevicePubKeysForPrimaryPubKey(primaryPubKey)
         // Don't send to ourselves
         .then(devicesPubKeys =>
           devicesPubKeys.filter(pubKey => pubKey !== ourNumber)
         )
         .then(devicesPubKeys => {
           if (devicesPubKeys.length === 0) {
```

Comment:
why send to ourselves? if everyone has it, do we really need to call `doSendMessage`?

Reply:
Not sure what you mean here, I just cleaned it up a bit, not changing the logic.

Reply:
yea, I see that but I think we should clean it up more. The first logic is to filter our own key (because we already have it) and then if there's no one to send it to, we send to ourselves... it's a bit weird or at least poorly documented.

the only thing I could be happening is that if you're a secondary device and you try to send it only to yourself, this would relay to primary key, but not even sure that makes any sense

Reply:
I believe you send to yourself in case you have a secondary device, in which case you would only send the message to that device. `doSendMessage` has to be called for public messages and group messages too. If there is a better way to express this, we could do it in the upcomig refactor.